src/Parser/Druid/Feral/Modules/FeralCore/Snapshot.js

```diff
@@ -42,30 +40,31 @@ class Snapshot extends Analyzer {
   };
 
   // extending class should fill these in:
-  spellCastId;
-  debuffId;
-  isProwlAffected;
-  isTigersFuryAffected;
-  isBloodtalonsAffected;
-  durationOfFresh;
+  spellCastId = null;
+  debuffId = null;
+  isProwlAffected = false;
+  isTigersFuryAffected = false;
+  isBloodtalonsAffected = false;
+  durationOfFresh = false;
 
   stateByTarget = {};
   lastDoTCastEvent;
 
+  castCount = 0;
+
   on_byPlayer_cast(event) {
     if (this.spellCastId !== event.ability.guid) {
       return;
     }
+    this.castCount += 1;
     this.lastDoTCastEvent = event;
-    
-    // attempt to associate with (very) recent debuff apply/refresh
-    // this cast could apply DoTs to multiple targets, so check them all
-    Object.values(this.stateByTarget).forEach(state => {
-      if (!state.castEvent && event.timestamp - state.startTime < CAST_WINDOW_TIME) {
-        event.castEvent = event;
-        this.checkRuleIfReady(state);
-      }
-    });
+  }
+
+  on_initialized() {
+    if (!this.spellCastId || !this.debuffId) {
+      this.active = false;
+      debug && console.warn('Snapshot missing spellCastId or debuffId.');
```

Comment:
This looks like a coding mistake, not something that can randomly occur. If that's the case I believe this should `throw new Error(...)` instead so it's more easily caught by developers.